OpenRA.Mods.Common/Traits/Player/BaseAttackNotifier.cs

```diff
@@ -44,6 +44,9 @@ public class BaseAttackNotifierInfo : TraitInfo
 		[Desc("Text notification to display to allies when under attack.")]
 		public readonly string AllyTextNotification = null;
 
+		[Desc("Trigger the notification for units only (instead of buildings only).")]
+		public readonly bool TriggerOnUnits = false;
```

Comment:
Wouldn't it be better to introduce some enum TriggerType: Building|Unit|Building&Unit

The name itself is not very clear.

Untested.